frostsnapp/lib/device_setup.dart

```diff
@@ -40,15 +41,23 @@ class DeviceSetup extends StatelessWidget {
   }
 }
 
+enum DeviceNameMode {
+  /// The name field renames the device and prompts the user for confirmation.
+  rename,
+
+  /// The name field stages the device name and persists it after keygen finalizes.
```

Comment:
Is this true (it may be)?
Noticed that we send `DeviceSendBody::SetName` in FinalizeKeyGen

Reply:
Possibly related:

After creating a wallet, if I immediately go to the devices list page, one of the devices is missing a name. But if i focus out of the app and back in, it becomes named.

Reply:
Good catch we probably are forgetting to update the device list when we finish keygen.

Reply:
> After creating a wallet, if I immediately go to the devices list page, one of the devices is missing a name. But if i focus out of the app and back in, it becomes named.

I don't think you need to focus out of the app for this to fix itself. Just wait a few seconds. This is because nonces are being gathered at this moment so it takes a little while for the devices to get their turn to set their name.